main/javascript-modifications/ses-guide.md

```diff
@@ -0,0 +1,506 @@
+undeniable objects. Many of these are identical in the new compartment.
+```js
+const c = new Compartment();
+c.globalThis === globalThis; // false
+c.globalThis.JSON === JSON; // true
+```
+
+Other pairs of compartments also share many identical intrinsics and undeniable objects
+of the realm. Each has a unique, initially mutable, global object.
+```js
+const c1 = new Compartment();
+const c2 = new Compartment();
+c1.globalThis === c2.globalThis; // false
+c1.globalThis.JSON === c2.globalThis.JSON; // true
+```
+Every compartment's global scope includes a shallow, specialized copy of the JavaScript
+intrinsics, omitting `Date.now()` and `Math.random()`. These are left out
+since they can be covert communication channels between programs. 
+
+However, a compartment may be expressly given access to these objects through
+the compartment constructor's first argument or by assigning them to the 
+compartment's `globalThis` after construction.
+```js
+const powerfulCompartment = new Compartment({ Math });
+powerfulCompartment.globalThis.Date = Date;
+```
+
+When a new `Compartment` object is created, you must decide if it supports object-capability security.
+If it does, run `harden(compartment.globalThis)` on it before loading any untrusted code into it.
+
+A single compartment can run a JavaScript program in the locked-down SES environment.
+However, most interesting programs have multiple modules. So, each compartment also has
+its own module system. The next SES release will include support for ECMAScript modules,
+**tyg todo: Has this happened already? If not, is there an ETA?**
+a relatively new system supported by many browsers, and officially released in Node.js 14.
+
+Compartments can be linked, so one compartment can export a module that another compartment
+imports. Each compartment may have its own rules for how to resolve import specifiers and 
+how to locate and retrieve modules. In this example, we use the compartment constructor to 
+create two compartments: one for the application and another for its dependency.
+
+**tyg todo: Will people understand this code or do we need to explain any/all methods in it
+in more detail**
+
+```js
+const dependency = new Compartment({}, {}, {
+  resolveHook: (moduleSpecifier, moduleReferrer) =>
+    resolve(moduleSpecifier, moduleReferrer),
+  importHook: async moduleSpecifier => {
+    const moduleLocation = locate(moduleSpecifier);
+    const moduleText = await retrieve(moduleLocation);
+    return new ModuleStaticRecord(moduleText, moduleLocation);
+  },
+});
+const application = new Compartment({}, {
+  'dependency': dependency.module('./main.js'),
+}, {
+  resolveHook,
+  importHook,
+});
+```
+
+Compartments greatly simplify using SES for large applications. We will build tools that use 
+compartments to load and bundle existing libraries and then execute them with a compartment
+for each package. **tyg todo Has this happened? If so, where/what are the tools? If not, any ETA?**
```

Comment:
This material was good for the blog but we should just omit it from the guide. This will be Endo.